odoo/tools/cache.py

```diff
@@ -100,9 +101,14 @@ def lookup(self, method, *args, **kwargs):
             counter.hit += 1
             return r
         except KeyError:
-            counter.miss += 1
-            value = d[key] = self.method(*args, **kwargs)
-            return value
+            with args[0].pool._Registry__caches_lock:
```

Comment:
Is it possible for a transaction has both `DROP TABLE` and ormcache usage?
Thread A holds the `_Registry__caches_lock` and waits for the `AccessShareLock` of the droped table
Thread B holds the `AccessExclusiveLock` for the dropped table and waits for the `_Registry__caches_lock`